src/main/java/net/minecraftforge/fml/common/network/handshake/NetworkDispatcher.java

```diff
@@ -257,7 +257,7 @@ else if (state != ConnectionState.CONNECTED && state != ConnectionState.HANDSHAK
         }
     }
 
-    private boolean handleVanilla(Packet msg)
+    private boolean handleVanilla(Packet<?> msg)
```

Comment:
I'm not sure you want this, as I haven't figured out how the whole class interacts with everything to generic type it properly, so I have left most network classes untouched.